test/inductor/test_triton_kernels.py

```diff
@@ -3127,9 +3127,13 @@ def branch_with_multiple_yield_args(
             {"ptr": t, "n_elements": 4, "BLOCK_SIZE": 4},
             ["ptr"],
         ],
-        # Cant optimize since the kernel contains a tl.inline_asm_elementwise
         [
-            inline_asm_kernel,
+            inline_asm_kernel_is_pure_true,
+            {"X": t, "Y": t, "Z": t, "n": 4, "BLOCK": 4},
+            ["Z"],
+        ],
```

Comment:
Do you have a test that shows we do not failure the mutation check on  inline_asm with `is_pure`?

Reply:
sry, i didn't catch it. You mean the test case of `is_pure` is set to `False`?

Reply:
Did this test case previously fail (raise a runtime error), or was it a perf issue?

If the test case previously failed without your PR, then I'm happy.

If the test case previously was just a perf issue (and not a correctness issue), then we may need a stronger test to prevent this from regressing.

Reply:
It's not the perf issue. It's a kernel with inline_asm would fail to compiling, reported in the user-group. After rounds of debugging, it pointed to the feature missing of handling inline_asm mutation analysis

Reply:
sounds good, thanks for clarifying